fw/apm.c

```diff
@@ -962,18 +966,33 @@ tfw_apm_calc(TfwApmData *data)
 	fn_unlock(&asent->rwlock);					\
 	pstats->seq = rdidx;						\
 									\
-	return (seq != rdidx);
+	seq != rdidx;							\
+})
 
 int
 tfw_apm_stats_bh(void *apmref, TfwPrcntlStats *pstats)
 {
-	__tfw_apm_stats_body(apmref, pstats, read_lock_bh, read_unlock_bh);
+	TfwApmData *data;
+	BUG_ON(!apmref);
+	data = &((TfwApmRef *)apmref)->data;
+	return __tfw_apm_stats_body(data, pstats, read_lock_bh, read_unlock_bh);
 }
 
 int
 tfw_apm_stats(void *apmref, TfwPrcntlStats *pstats)
 {
-	__tfw_apm_stats_body(apmref, pstats, read_lock, read_unlock);
+	TfwApmData *data;
+	BUG_ON(!apmref);
+	data = &((TfwApmRef *)apmref)->data;
+	return __tfw_apm_stats_body(data, pstats, read_lock, read_unlock);
+}
+
+int
+tfw_apm_stats_global(TfwPrcntlStats *pstats)
+{
+	BUG_ON(!tfw_apm_global_data);
```

Comment:
Just to clarify -- If this is to imitate whatever is done in the macro, then `apmref` is verified for non-zero value because it's dereferenced later. The resulting `data` is then dereferenced as well, but it's presumed set at this stage.